api_tests/base/test_sloan_study.py

```diff
@@ -0,0 +1,168 @@
+import pytest
+import mock
+
+from waffle.models import Flag
+
+from osf_tests.factories import (
+    AuthUserFactory,
+    PreprintFactory
+)
+
+from osf.features import (
+    SLOAN_DATA,
+    SLOAN_PREREG,
+    SLOAN_COI
+)
+
+SLOAN_FLAGS = (
+    SLOAN_DATA,
+    SLOAN_PREREG,
+    SLOAN_COI
+)
+
+
+@pytest.mark.django_db
+class TestSloanStudyWaffling:
```

Comment:
how about a test with mixed flags?

 e.g. user has one false cookie, one true cookie, and one missing; user tags similarly mixed; check that they combine correctly